### DISCUSSION TITLE
Using Denon AVR as Splitter

### DISCUSSION THREAD

**Discussion by definitelynotpaul** [awawa-dev/HyperHDR#963] [Q&A]
I recently switched from a WLED/Android Grabber setup to HyperSerialPico/RPi4b (thank you for the help @awawa-dev). Everything seems to be working as intended from HyperHDR with much better performance than before. I am running into some frustrating behavior however. Here's my hardware setup:
Host: RPi 4b running SD Card image
Driver: RP2040 Scorpio
Strip: SK6812 Cool White
Splitter: Denon AVR S970H
Capture: MS2130 Model B reflashed with f3 customer firmware

After flashing with the custom firmware and downloading the LUT, colors in SDR content with HDR tone mapping off appear to be very accurate. The problem I'm running into is that while outputting to both the TV and MS2130 via "Monitor 1/ Monitor 2" the AVR will not output an HDR signal. I believe it has something to do with this section of the manual:
![image](https://github.com/user-attachments/assets/47610976-8a3c-435f-acfb-eca5316f996d)

I have tried with both the upscaler off and set to auto, but I'm thinking the MS2130 is not recognized as capable of receiving DV/HDR10 and so the AVR is sending 4K SDR. Did I do something wrong when flashing the custom firmware, or if this is correct that even with the f3 firmware, the MS2130 will not accept DV/HDR10, is there a different grabber (other than Elgato with it's built in splitter) that will work from Output 2 of the AVR?

**Comment by awawa-dev** [comment]
Hi
The splitters built into Denon receivers are a joke - they are not fully functional HDMI splitters if we compare them to specialized splitters available on the market, they lack many features and are problematic for more advanced applications and usually make them impossible.

**Comment by definitelynotpaul** [comment]
Here's an update. I'll be returning the HD60X. Unfortunately no matter where in-line I put the HD60X it will not suit my needs. 
Here's what I tried:
Scenario 1
`Shield> AVR input > Output 1/2
                    Output 1 > TV
                    Output 2 > HD60X's input > HD60X's output unplugged (this does not effect it's ability to send screencap to HyperHDR)`
In this configuration (I assume) the AVR is unable to send a constant signal out, for whatever reason so I get 1 second of video followed by 1 second of black screen indefinitely.
Scenario 2
`Shield Pro > AVR input >AVR Output 1 > HD60X input > HD60X output > TV`
Same result as first scenario.

Scenario 3
'Shield Pro > HD60X input > HD60X output > AVR input > AVR output 1 > TV'
This scenario was very promising. I was able to get a solid output including proper HDR. Everything looked good from the LEDs. However, this is when I discovered the HD60X does not passthrough HDMI audio. It also does not strip HDCP. This defeats my AVR entirely.

At this point I need some advice on a splitter that will allow me to retain HDR/DV passthrough, DTS/True-HD audio passthrough, CEC, eARC (this is not a dealbreaker). @awawa-dev any suggestions?

**Reply by awawa-dev** [reply]
No licensed grabber or so-called "splitter" built into Denon will support HDCP "workaround" ever. And the latter usually also lacks a decent scaler that can convert resolution and color depth at once. I feel a bit sorry you're giving up on HD60X because it offers color quality that's hard to beat (it's only beaten by av access with our HDR tone mapping, but for a different reason: https://github.com/awawa-dev/HyperHDR/pull/896#issuecomment-2427827710). Usually users use Ezcoo EZ-SP12H2 or its relative with 4k120 support but both dont support for example CEC.

**Reply by definitelynotpaul** [reply]
Honestly, I'd love to keep it, I just don't understand why I'm getting this signal issue. If I run directly from the Shield I have no issues. I'm going to try today connecting it that way to TV Input 2 and run eARC to the Denon and see if that works. I'm not sure if it will pass audio that way. 

Can you think of a way I can diagnose exactly why I'm having connectivity issues running through the Denon?

**Reply by awawa-dev** [reply]
I don't know exactly what the problem might be, maybe contact Denon or Elgato support? From my experience with my Denon AVR I have found that this built-in "splitter" is completely useless, at least for me. And that was even before I bought the Elgato...

**Reply by awawa-dev** [reply]
If you really want to experiment (without any guarantee of success) and have the option to return the equipment if it doesn't work, you can try:
Shield> AVR input > Output 1/2 Output 1 > TV, Output 2 > Ezcoo splitter with copied EDID from TV -> **Ezcoo output 2**-> HD60X's input > HD60X's output unplugged, if Shield is using HDCP then AVR will cut off the signal anyway

**Comment by elco1984** [comment]
Well this is a bummer. I too have a Denon receiver, AVR X2100W, I have tried 2 different capture cards. I have HyperHDR installed on a RPI4. I had Inconsistent success the card linked below. It would work and the stop working for no apparent reason. 
https://www.amazon.com/gp/product/B0CDLH7DMK/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1

Then after some reading I decided to purchase the Elgato HD60X. I am also using the AVR as a HDMI switch as I have multiple sources. DirecTV, Odroid running CoreElec, and a Chromecast. I have had similar failed results as definaletynotpaul when using the HD60X. 

If the Denon HDMI switch is the issue, other than purchasing a new AVR how would one prodeceed?  


Thanks

**Comment by awawa-dev** [comment]
@elco1984 what exactly is the problem in your case? BTW Elgato completely replaced the HD60X, its internal layout and behavior are completely different, check if you have the new model (I have had it for a few days and returned it) by the product id and vendor id. In the case of the new model it is hard for me to advise anything More details: https://github.com/awawa-dev/HyperHDR/discussions/1010

**Comment by elco1984** [comment]
The problem I am having when I use the HD60X, I get no video out of the either HDMI outs from the receiver. Like deffinenatlynotpual's post above I have put the HD60X in different places in the system and I get no video signal. I am not sure witch version I have. I will try and confirm when I get home tonight.

**Reply by awawa-dev** [reply]
Are you trying to use Denon as a splitter? Rather bad idea. First make sure there is no HDCP anywhere in the HDMI link.

**Comment by elco1984** [comment]
I was attempting to use the Denon as a splitter. Until I found this post and read your comment above. How would one proceed with a Denon AVR? I will have to check the HDMI specs on all the equipment tonight. The newest piece of equipment I have is the TV. I just bought a Samsung s90. I am sure it is HDCP 2.2. 

Looking at the post you linked above, where does one find the vendor and product id? Looking at Hyperhdr, video capturing, device info, I see video 0. Looking at overview information it shows the same. Thanks for the quick replies

**Reply by awawa-dev** [reply]
You should find product/vendor_id in the device manager (Windows) somewhere in the device properties, or lsusb in Linux if I remember correctly. 

If HDMI is HDCP protected on the source (the TV only adapts) and you can't turn it off, don't count on this substitute of splitter called the Denon AVR to let you split the signal. To bypass HDCP, you need a real splitter like Ezcoo. Of course, we're still talking about the case that HDCP is active.

**Reply by elco1984** [reply]
It appears as if I have the new model USB\VID_0FD9&PID_008A&MI_00\6&d91d4c5&0&0000. As far as HDCP is concerned the
Denon AVR X2100W does not support it, Directv HR24 does not have HDCP2.2 support. And I can't find any evidence that the Odroid N2+ has HDCP support. The Chromecast Ultra does, I think.

**Reply by awawa-dev** [reply]
the key is the video source and for ex. 4K must support HDCP 2.2, maybe Denon cannot handle the signal or some its settings must be changed, dont know. Despite I have Denon 2700 I absolutely don't recommend to use it as a splitter, it doesnt handle hdcp and scalling for each of the output.

**Reply by elco1984** [reply]
Understood. I suspect you are correct about Denon being the issue. I have turned off HDMI scaling and tried other settings changes in the Denon. 

So you run all sources to a splitter then to your Denon and capture card? Essentially the Denon is only handling audio.

**Reply by elco1984** [reply]
If by source you mean the material I'm streaming I can tell you it's not 4k. Event the local media I stream from a server is not 4k.

**Reply by awawa-dev** [reply]
My PC GFX has dual output, one only for the audio to the Denon and the second for video then to the grabber with a loop (previously Elgato HD60X, currently latest UGreen for P010) then TV. I disabled HDCP in my AMD gfx.

**Reply by definitelynotpaul** [reply]
Sorry for not updating this thread earlier. My setup is working flawlessly using the Denon (970H) as the splitter. Not with the HD60X however. Both HDCP and DV ended up being deal breakers for me. I am capturing using the recommended MS2130 from AliExpress flashed with @awawa-dev custom firmware.

I don't remember my output settings on the Denon that made everything work, or if they match your model, but if you need them I can post them later on.

**Reply by elco1965** [reply]
@definitelynotpaul I have ordered the MS2130. If you have found the recipe to the Denon secret sauce, I would appreciate you sharing. Even if the settings are not identical between your 970H and my X2100W I hope they will get my going in right direction. 

Thanks for the help

**Reply by pho084** [reply]
> Sorry for not updating this thread earlier. My setup is working flawlessly using the Denon (970H) as the splitter. Not with the HD60X however. Both HDCP and DV ended up being deal breakers for me. I am capturing using the recommended MS2130 from AliExpress flashed with @awawa-dev custom firmware.
> 
> I don't remember my output settings on the Denon that made everything work, or if they match your model, but if you need them I can post them later on.

Sorry for bringing this up again  but your post confused me :D
I am currently looking at the 970h for that exact purpose. Do you have it working reliably by now with the ms2130? And does it work for hdr/dv and hdcp content in 4k?
Thanks
